app/routes/romm.py

```
@@ -0,0 +1,1063 @@
+    skipped = 0
+    # Source -> destination for every row written, returned so the caller can
+    # retire the ones its batch does not end up queueing. Keyed by source
+    # because that is what the caller submits and what the created jobs report
+    # back; the destination is what a row is identified by.
+    recorded_paths: dict[str, str] = {}
+    for path in payload.paths:
+        if not is_within_configured_volumes(path):
+            skipped += 1
+            continue
+        # Same canonical key the index was built with (see roms_by_local_path).
+        rom = platform_roms.get(os.path.realpath(path))
```

Ruff (0.16.1)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)